src/BottomNavigation/BottomNavigationItem.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import shallowEqual from 'recompose/shallowEqual';
+import EnhancedButton from '../internal/EnhancedButton';
+
+function getStyles(props, context) {
+  const {selected} = props;
+
+  const {muiTheme} = context;
+  const {bottomNavigation} = muiTheme;
+
+  const color = selected ? bottomNavigation.selectedColor :
+                           bottomNavigation.unselectedColor;
+  const styles = {
+    root: {
+      transition: 'all 0.3s',
```

Comment:
I considered the same thing, as I typically find `transition: all` really annoying due to its tendency to randomly transition the wrong properties and it leaves no evidence of the properties that were actually supposed to transition.

However, it is possible to pass an arbitrary element as an icon, which could have any number of properties that need transitioning. This occurred for me in my app, so I resolved to keep it as `all`. What do you think?